### PR TITLE
UNR-5679: PushModel NetDeltaSerialized properties warning

### PR DESCRIPTION
**Pull request by improbable-dmitrii**
#### Description
This only adds warnings when PushModel is active and a class containing a NetDeltaSerialized property is replicated; no new functionality added. Follow-up on #3327.

#### Tests
Run the functional test suite.

#### Documentation
Added a changelog entry.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp [hunk 1]
```diff
@@ -157,6 +157,11 @@ void USpatialClassInfoManager::CreateClassInfoForClass(UClass* Class)
 		return;
 	}
 
+	Class->SetUpRuntimeReplicationData();
+
+	TArray<FLifetimeProperty> LifetimeReplicatedProperties;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp:162]: #include "UObject/CoreNet.h" ?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp [hunk 2]
```diff
@@ -157,6 +157,11 @@ void USpatialClassInfoManager::CreateClassInfoForClass(UClass* Class)
 		return;
 	}
 
+	Class->SetUpRuntimeReplicationData();
```
[the remark this answers is not in the document]
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp:160]: On I see at that point it's wrapped in a for loop. I'm in favour of centralising this code, and just iterating over the class properties a second time within some push model checks.

File: SpatialGDK/Source/SpatialGDK/SpatialGDK.Build.cs [hunk 3]
```diff
@@ -40,6 +40,11 @@ public SpatialGDK(ReadOnlyTargetRules Target) : base(Target)
                 "UMG"
             });
 
+        if (Target.bWithPushModel)
+        {
+            PublicDependencyModuleNames.Add("NetCore");
```
[the remark this answers is not in the document]
Comment (improbable-dmitrii) [reply SpatialGDK/Source/SpatialGDK/SpatialGDK.Build.cs:45]: `IS_PUSH_MODEL_ENABLED` https://github.com/spatialos/UnrealGDK/pull/3337/files#diff-07317e4696ba4273343ce0815317410a5a35e1b37bb8381186025c82a8e32bfeR232 expands to `UE4PushModelPrivate::IsPushModelEnabled()` which uses an `extern NETCORE_API bool bIsPushModelEnabled`, so we need to import that.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (improbable-dmitrii) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp:160]: > Performance concern here?
It's amortized, the method exits early if it's called for the second time on an actor class.

[the remark this answers is not in the document]
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp:160]: Ahh glazed over `CLASS_ReplicationDataIsSetUp` assuming it was static for the lifetime of the class. Fine 👍

**Review by m-samiec — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp [hunk 4]
```diff
@@ -223,6 +224,41 @@ void USpatialClassInfoManager::CreateClassInfoForClass(UClass* Class)
...
+	if (IS_PUSH_MODEL_ENABLED() && GetDefault<USpatialGDKSettings>()->bShouldWarnOnNetDeltaSerializedPushModel)
+	{
+		Class->SetUpRuntimeReplicationData();
+
+		TArray<FLifetimeProperty> LifetimeReplicatedProperties;
+		Class->GetDefaultObject()->GetLifetimeReplicatedProps(LifetimeReplicatedProperties);
+
+		for (TFieldIterator<FProperty> PropertyIt(Class); PropertyIt; ++PropertyIt)
+		{
+			if (FStructProperty* StructProperty = CastField<FStructProperty>(*PropertyIt))
```
Comment (samiwh) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp:237]: Nit - can we invert any of these. There's a lot nesting going on.

**Review by samiwh — APPROVED** [review]
Discussed a couple of small things offline but otherwise lgtm

**Pull request merged**